__tests__/licenses.test.ts

```diff
@@ -68,3 +68,31 @@ test('it fails all license checks when allow is provided an empty array', async
   })
   expect(invalidChanges.length).toBe(2)
 })
+
+test('it does not fail if a license outside the allow list is found in removed changes', async () => {
+  const changes: Changes = [
+    {...npmChange, change_type: 'removed'},
+    {...rubyChange, change_type: 'removed'}
+  ]
+  const [invalidChanges, _] = getDeniedLicenseChanges(changes, {allow: ['BSD']})
+  expect(invalidChanges).toStrictEqual([])
+})
+
+test('it does not fail if a license inside the deny list is found in removed changes', async () => {
+  const changes: Changes = [
+    {...npmChange, change_type: 'removed'},
+    {...rubyChange, change_type: 'removed'}
+  ]
+  const [invalidChanges, _] = getDeniedLicenseChanges(changes, {deny: ['BSD']})
+  expect(invalidChanges).toStrictEqual([])
+})
+
+test('it fails if a license outside the allow list is found in both of added and removed changes', async () => {
+  const changes: Changes = [
+    {...npmChange, change_type: 'removed'},
+    npmChange,
```

Comment:
I don't understand in which case we could get both an added/removed change, but I like the idea of sticking to what's specified in the allowlist if that happens.

Reply:
Sorry for my laziness of creating rough fixtures 🙇‍♂️ I think same package does not appear in both added/removed. However, it might occur in different packages.

```zsh
gh api \
  -H "Accept: application/vnd.github+json" \
  /repos/fastify/fastify-passport/dependency-graph/compare/4b912074ad2273172b65eb44e52201e37583226e...62bdea61e7c137028263b59943de4e65583db760 | jq '.[] | {name: .name, change_type: .change_type}'
```

```json
{
  "name": "@fastify/fast-json-stringify-compiler",
  "change_type": "added"
}
{
  "name": "fast-json-stringify",
  "change_type": "added"
}
{
  "name": "fastify",
  "change_type": "added"
}
{
  "name": "find-my-way",
  "change_type": "added"
}
{
  "name": "@fastify/fast-json-stringify-compiler",
  "change_type": "removed"
}
{
  "name": "fast-json-stringify",
  "change_type": "removed"
}
{
  "name": "fastify",
  "change_type": "removed"
}
{
  "name": "find-my-way",
  "change_type": "removed"
}
{
  "name": "string-similarity",
  "change_type": "removed"
}
```